### PR TITLE
agent: some cleanup

### PR DESCRIPTION
**Pull request by lcaw**
We could use tidy up the error handling a bit and remove
get_vsock_incoming() entirely to save a few lines.

Fixes: #1650

Signed-off-by: Qian Cai <qcai@redhat.com>

### CODE REVIEW

File: src/agent/src/main.rs [hunk 1]
```diff
@@ -110,21 +110,20 @@ fn announce(logger: &Logger, config: &AgentConfig) {
 
 fn set_fd_close_exec(fd: RawFd) -> Result<RawFd> {
     if let Err(e) = fcntl::fcntl(fd, FcntlArg::F_SETFD(FdFlag::FD_CLOEXEC)) {
-        return Err(anyhow!("failed to set fd: {} as close-on-exec: {}", fd, e));
+        bail!("failed to set fd: {} as close-on-exec: {}", fd, e);
```
Comment (liubin) [inline src/agent/src/main.rs:113]: This is simple, but I'm confused if should we replace all `return Err(anyhow!())` to `bail!` in all the rust files.

Comment (lcaw) [reply src/agent/src/main.rs:113]: Until someone objections, I'll send another PR later to do tree-wide changes for this.

Comment (Tim-Zhang) [reply src/agent/src/main.rs:113]: I agree with @liubin, I think we should replace all `return Err(anyhow!)` to `bail!`, or we don't change any of it.

Comment (Tim-Zhang) [reply src/agent/src/main.rs:113]: FYI: I have remove this function `set_fd_close_exec ` in another pr: https://github.com/kata-containers/kata-containers/pull/1648

Comment (lcaw) [reply src/agent/src/main.rs:113]: > I agree with @liubin, I think we should replace all `return Err(anyhow!)` to `bail!`, or we don't change any of it.

Okay, I'll resubmit to defer this change for this PR.

Comment (lcaw) [reply src/agent/src/main.rs:113]: > FYI: I have removed this function `set_fd_close_exec ` in another pr: #1648

Okay, I'll rebase on it once your PR get merged (hopefully tomorrow).

Comment (Tim-Zhang) [reply src/agent/src/main.rs:113]: thanks a lot

**Comment by lcaw** [comment]
Don't need this at the moment due to recent commits.

**Pull request closed without merging**